commcare_connect/form_receiver/processor.py

```diff
@@ -230,6 +230,18 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
                 if dist.m <= 10:
                     flags.append(["location", "Visit location is too close to another visit"])
                     break
+        if opportunity_flags.catchment_areas:
+            areas = access.catchmentarea_set.filter(active=True)
+            if areas:
+                cur_lat, cur_lon, *_ = xform.metadata.location.split(" ")
```

Comment:
It might be worth making this a separate method outside of `process_deliver_unit`